checker/src/org/checkerframework/checker/samelen/SameLenTransfer.java

```diff
@@ -0,0 +1,134 @@
+            ArrayCreationNode acNode = (ArrayCreationNode) node.getExpression();
+            if (acNode.getDimensions().size() == 1 && isArrayLengthAccess(acNode.getDimension(0))) {
+                // This array that's being created is the same size as the other one.
+                AnnotationMirror combinedSameLen =
+                        aTypeFactory.createCombinedSameLen(
+                                ((FieldAccessNode) acNode.getDimension(0)).getReceiver().toString(),
+                                node.getTarget().toString(),
+                                aTypeFactory
+                                        .getAnnotatedType(
+                                                ((FieldAccessNode) acNode.getDimension(0))
+                                                        .getReceiver()
+                                                        .getTree())
+                                        .getAnnotationInHierarchy(UNKNOWN),
+                                aTypeFactory.createSameLenUnknown());
+
+                Receiver targetRec =
+                        FlowExpressions.internalReprOf(analysis.getTypeFactory(), node.getTarget());
+                Receiver otherRec =
+                        FlowExpressions.internalReprOf(
+                                analysis.getTypeFactory(),
+                                ((FieldAccessNode) acNode.getDimension(0)).getReceiver());
+
+                result.getRegularStore().clearValue(targetRec);
+                result.getRegularStore().insertValue(targetRec, combinedSameLen);
+                result.getRegularStore().clearValue(otherRec);
+                result.getRegularStore().insertValue(otherRec, combinedSameLen);
+            }
+        }
+
+        return result;
+    }
+
+    private boolean isArrayLengthAccess(Node node) {
+        return (node instanceof FieldAccessNode
+                && ((FieldAccessNode) node).getFieldName().equals("length")
+                && ((FieldAccessNode) node).getReceiver().getType().getKind() == TypeKind.ARRAY);
+    }
+    /**
+     * Handles refinement of equality comparisons. Looks for a.length and b.length, then annotates
+     * both a and b to have SameLen of each other in the store.
+     */
+    private void refineEq(Node left, Node right, CFStore store) {
+        if (isArrayLengthAccess(left)) { // FIXME add list support here
+            if (isArrayLengthAccess(right)) {
+                Receiver leftRec =
+                        FlowExpressions.internalReprOf(
+                                analysis.getTypeFactory(), ((FieldAccessNode) left).getReceiver());
+                Receiver rightRec =
+                        FlowExpressions.internalReprOf(
+                                analysis.getTypeFactory(), ((FieldAccessNode) right).getReceiver());
+                AnnotationMirror combinedSameLen =
+                        aTypeFactory.createCombinedSameLen(
```

Comment:
This is impossible to read. Please extract some of the arguments to local variables.